### PR TITLE
Allow syslog over TCP+TLS.

### LINKED ISSUE #18364: Replace Golang log/syslog pkg because it's no longer actively developed
Following the discussion on https://github.com/docker/docker/issues/18358, it appears that Golang has decided to stop development of the log/syslog package;

From https://github.com/golang/go/issues/13449;

>We have no plans to further develop the log/syslog package in the standard library.
> We welcome the community to fork it and give it features elsewhere.
> It was probably a mistake putting it in the standard library to begin with.

We should look for another package to replace it.

### PR DESCRIPTION
**Pull request by calavera**
Replace Go's stdlib version because it's no longer maintained.

Fixes #18364

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks @calavera!

**Comment by calavera** [comment]
This is blocked by https://github.com/RackSec/srslog/pull/5.

**Comment by thaJeztah** [comment]
Should we close this in the meantime?

**Comment by calavera** [comment]
> Should we close this in the meantime?

no need, it was just merged. I'm updating the vendor.

**Comment by icecrime** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Comment by cpuguy83** [comment]
Actually, can you split the vendor into a separate commit?

**Comment by runcom** [comment]
also needs a a rebase :(

**Comment by calavera** [comment]
@cpuguy83, @runcom rebased and split

**Comment by runcom** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Comment by LK4D4** [comment]
ping @thaJeztah @moxiegirl for review

File: docs/reference/logging/overview.md [hunk 1]
```diff
@@ -107,6 +111,14 @@ the following named facilities:
 * `local6`
 * `local7`
 
+`syslog-tls-ca-cert` specifies the absolute path in the host to the Trust certs signed by the CA. This option is ignored if the address protocol is not `tcp+tls`.
```
Comment (thaJeztah) [inline docs/reference/logging/overview.md]: minor nits;
s/Trust/trust/
s/certs/certificates/


also, can you wrap your lines to 80 chars?

Comment (thaJeztah) [inline docs/reference/logging/overview.md]: oh, and absolute path in the host -> on the host? (or simply "the absolute path")

**Comment by thaJeztah** [comment]
Thanks @calavera; some minor nits

**Comment by calavera** [comment]
@thaJeztah I addressed your comments.

**Comment by vdemeester** [comment]
docs LGTM 🐉

File: docs/reference/logging/overview.md [hunk 2]
```diff
@@ -107,6 +111,19 @@ the following named facilities:
 * `local6`
 * `local7`
 
+`syslog-tls-ca-cert` specifies the absolute path to the trust certificates
```
Comment (jamtur01) [inline docs/reference/logging/overview.md]: Maybe form each of these into proper sentences:

  The `syslog-tls-ca-cert` options specifies ....

**Comment by jamtur01** [comment]
Docs LGTM minor comment.

**Comment by thaJeztah** [comment]
@jamtur01 agreed, it can use a touch-up; I compared it to the existing docs, and decided that it's probably better to do that in a separate PR. Thanks for helping out reviewing James, much appreciated!!

LGTM

I'll merge this; we can revisit in e separate PR

**Pull request merged**